File: Generacidad/Ej3/Catalogo.java
package Generacidad.Ej3;

import java.util.ArrayList;

public class Catalogo<T> {
    private ArrayList<T> elementos = new ArrayList<>();

    public void agregar(T elemento) {
        elementos.add(elemento);
    }

    public T buscar(int index) {
        if (index >= 0 && index < elementos.size()) {
            return elementos.get(index);
        }
        return null;
    }

    public static void main(String[] args) {
        // Catálogo de Libros
        Catalogo<String> catalogoLibros = new Catalogo<>();
        catalogoLibros.agregar("El Principito");
        catalogoLibros.agregar("Cien años de soledad");
        System.out.println("Libro encontrado: " + catalogoLibros.buscar(1));

        // Catálogo de Productos
        Catalogo<Double> catalogoPrecios = new Catalogo<>();
        catalogoPrecios.agregar(12.99);
        catalogoPrecios.agregar(8.50);
        System.out.println("Precio encontrado: " + catalogoPrecios.buscar(0));
    }
}

File: Generacidad/Ej3/Catalogo.py
class Catalogo:
    def __init__(self):
        self.elementos = []

    def agregar(self, elemento):
        self.elementos.append(elemento)

    def buscar(self, index):
        if 0 <= index < len(self.elementos):
            return self.elementos[index]
        return None

catalogo_libros = Catalogo()
catalogo_libros.agregar("El Principito")
print(f"Libro encontrado: {catalogo_libros.buscar(0)}")

catalogo_precios = Catalogo()
catalogo_precios.agregar(19.99)
print(f"Precio encontrado: {catalogo_precios.buscar(0)}")
